File: 1657-determine-if-two-strings-are-close/1657-determine-if-two-strings-are-close.cpp
class Solution {
public:
    bool closeStrings(string word1, string word2) {
        vector<int> v1(26, 0);
        vector<int> v2(26, 0);
        vector<int> v11(26, 0);
        vector<int> v22(26, 0);
        
        for(auto i : word1){
            v1[i-'a']++;
            v11[i-'a'] = 1;
        }
        
        for(auto i : word2){
            v2[i-'a']++;
            v22[i-'a'] = 1;
        }
        
        sort(v1.begin(), v1.end());
        sort(v2.begin(), v2.end());
        
        
        return v1==v2 && v11==v22;
    }
};

File: 1657-determine-if-two-strings-are-close/1657-determine-if-two-strings-are-close.py
class Solution:
    def closeStrings(self, word1: str, word2: str) -> bool:
        if len(word1) != len(word2):
            return False
        
        freq1 = Counter(word1)
        freq2 = Counter(word2)
        
        v1 = sorted([i for i in freq1.values()])
        v2 = sorted([i for i in freq2.values()])
        
        return v1==v2 and freq1.keys()==freq2.keys()
